android/pytorch_android/src/androidTest/java/org/pytorch/PytorchInstrumentedTests.java

```diff
@@ -258,6 +258,46 @@ public void testTensorIllegalStateOnWrongType() {
     tensorFloats.getDataAsByteArray();
   }
 
+
+  @Test
+  public void testEqString() throws IOException {
+    final Module module = Module.load(assetFilePath(TEST_MODULE_ASSET_NAME));
+    String[] values =
+        new String[] {
+            "smoketest",
+            "проверка не латинских символов", // not latin symbols check
+            "#@$!@#)($*!@#$)(!@*#$"
+        };
+    for (String value : values) {
+      final IValue input = IValue.string(value);
+      assertTrue(input.isString());
+      assertTrue(value.equals(input.getString()));
```

Comment:
This is fine, but I would consider upgrading the test suite to use fluent assertions (with Truth or AssertJ, see https://truth.dev/comparison.html) in a separate diff.  If this test fails for whatever reason, you'll just get an error that says "expected false to be true".  With Truth, you'd write `assertThat(input.getString()).isEqualTo(value)` and get a proper error that showed the two strings being compared.  The benefits are even more significant when doing more complex assertions like substrings, regex, or container contents.

Reply:
Thanks!
I plan to add more tests next week and will integrate those assertions.